packages/jellyfish-transaction-builder/src/txn/txn_builder_masternode.ts

```diff
@@ -41,4 +48,87 @@ export class TxnBuilderMasternode extends P2WPKHTxnBuilder {
       changeScript
     )
   }
+
+  /**
+   * Build update masternode transaction
+   *
+   * @param {UpdateMasternode} updateMasternode transaction to create
+   * @param {Script} changeScript to send unspent to after deducting the (converted + fees)
+   * @param {Object} [collateralInfo] needed when updating owner address
+   * @param {RawTransaction} collateralInfo.rawCollateralTx to get relevant data needed for collateral vin and vout
+   * @param {Script} collateralInfo.newOwnerScript for new owner address
+   * @return {Promise<TransactionSegWit>}
+   */
+  async update (
+    updateMasternode: UpdateMasternode,
+    changeScript: Script,
+    collateralInfo?: {
+      rawCollateralTx: RawTransaction
+      newOwnerScript: Script
+    }
+  ): Promise<TransactionSegWit> {
+    const minFee = new BigNumber(0.001)
+    const { prevouts, vin, total } = await this.collectPrevouts(minFee)
+
+    const deFiOut: Vout = {
+      value: new BigNumber(0),
+      script: {
+        stack: [
+          OP_CODES.OP_RETURN,
+          OP_CODES.OP_DEFI_TX_UPDATE_MASTER_NODE(updateMasternode)
+        ]
+      },
+      tokenId: 0x00
+    }
+
+    const change: Vout = {
+      value: total,
+      script: changeScript,
+      tokenId: 0x00
+    }
+
+    const mergedVin = [...vin]
+    const mergedVout = [deFiOut, change]
+    const mergedPrevouts = [...prevouts]
+
+    if (collateralInfo != null) {
```

Comment:
should prefer strict equality checking (` !==` )

Reply:
amended